pkg/products/threescale/reconciler.go

```diff
@@ -2279,11 +2279,19 @@ func (r *Reconciler) reconcileRatelimitingTo3scaleComponents(ctx context.Context
 		ApicastContainerPort,
 	)
 
+	// apicast filters based on installation type
+	apicastHTTPFilters, err := getAPICastHTTPFilters(installation)
```

Comment:
instead of passing the installation type to the function would verify the install type in the reconciler and then call `getAPICastHTTPFilters` and `getAPICastHTTPFiltersWithHeaders` it would remove the dependency of the installation type on another go file